.github/workflows/test.yml

```diff
@@ -2,55 +2,57 @@ name: Test
 
 on:
   push:
-    branches: [main]
   pull_request:
-    branches: [main]
+  schedule:
+    # Run monthly to keep Rust toolchain changes fresh
+    - cron: '0 0 1 * *'
 
 jobs:
   rustfmt:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        channel: [stable, beta]
     steps:
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@master
         with:
-          toolchain: stable
-          override: true
+          toolchain: ${{ matrix.channel }}
           components: rustfmt
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Run `cargo fmt`
-        uses: actions-rs/cargo@v1
-        with:
-          command: fmt
-          args: --all -- --check
+        run: |
+          cargo fmt --all -- --check
 
   clippy:
     needs: rustfmt
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        channel: [stable, beta]
     steps:
-      - uses: actions-rs/toolchain@v1
+      - uses: dtolnay/rust-toolchain@master
         with:
-          toolchain: stable
-          override: true
+          toolchain: ${{ matrix.channel }}
           components: clippy
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Run `cargo clippy`
-        uses: actions-rs/clippy-check@v1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          cargo clippy -- -D warnings
```

Comment:
Please remove `-D warnings`, because I don't want the workflow to break when a new warning is added to clippy.

Reply:
This part of the job doesn't really do much without `-D warnings` unless someone sifts through the CI logs. I know `actions/clippy-check` would leave github review comments, but I don't know of a maintained alternative

If you want I can switch it to do a `cargo check` instead of `cargo clippy`

Reply:
I think `clippy` is still useful because it has about 70 lints that are deny-by-default.

Reply:
I didn't think about that. Good point!

`-D warnings` has been removed